tracing/tracer.py

```diff
@@ -164,16 +164,21 @@ def _on_trace_is_called(self, frame, event, arg: typing.Any) -> typing.Callable:
         line_number = frame.f_lineno
 
         names2types, category = None, None
+        frameinfo = inspect.getframeinfo(frame)
+
         if event == "call":
-            logger.info(f"Tracing call: {inspect.getframeinfo(frame)}")
+            logger.info(f"Tracing call: {frameinfo}")
             names2types = self._on_call(frame, arg)
             category = TraceDataCategory.FUNCTION_ARGUMENT
 
         elif event == "return":
-            logger.info(f"Tracing return: {inspect.getframeinfo(frame)}")
+            logger.info(f"Tracing return: {frameinfo}")
             names2types = self._on_return(frame, arg)
             category = TraceDataCategory.FUNCTION_RETURN
 
+            # Special case
+            line_number = 0
```

Comment:
Remove these lines.

Reply:
No, they are needed in the later call to update the data frame. it overwrites the number taken from the frame